src/Components/Browser.JS/src/Platform/Circuits/ComponentDescriptor.ts

```diff
@@ -0,0 +1,43 @@
+import { attachRootComponentToLogicalElement } from '../../Rendering/Renderer';
+import { toLogicalRootCommentElement } from '../../Rendering/LogicalElements';
+
+export interface EndComponentComment {
+  componentId: number;
+  node: Comment;
+  index: number;
+}
+
+export interface StartComponentComment {
+  node: Comment;
+  rendererId: number;
+  componentId: number;
+  circuitId: string;
+}
+
+// Represent pairs of start end comments indicating a component that was registered
+// in markup (such as a prerendered component)
+export interface MarkupRegistrationTags {
+  start: StartComponentComment;
+  end: EndComponentComment;
+}
+
+export class ComponentDescriptor {
+  public registrationTags: MarkupRegistrationTags;
```

Comment:
Do we definitely need to track `registrationTags`, or even have the `StartComponentComment`/`EndComponentComment` types? Once L40 has built a logical element, isn't that the only thing we need to keep hold of? That already defines the exact region of the document where output will later go, as well as what existing nodes will be cleared out.

Reply:
Oh wait, I see - you need to track the set of circuit IDs so you can put them into groups.

Reply:
You are right, but most of this information is useless after that. I don't think it's a biggie